src/sources/logplex.rs

```diff
@@ -17,22 +18,73 @@ use std::{
 };
 use warp::http::{HeaderMap, StatusCode};
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct Auth {
+    username: String,
+    password: String,
+}
+
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct LogplexConfig {
     address: SocketAddr,
     tls: Option<TlsConfig>,
+    auth: Option<Auth>,
 }
 
 inventory::submit! {
     SourceDescription::new_without_default::<LogplexConfig>("logplex")
 }
 
 #[derive(Clone, Default)]
-struct LogplexSource {}
+struct LogplexSource {
+    auth: Option<String>,
+}
 
 impl HttpSource for LogplexSource {
```

Comment:
Hmm, so this adds basic auth to the logplex source, but not all http sources? Would it be much harder to make the implementation generic?

Reply:
Good suggestion @Hoverbear! I didn't thought about generic implementation, that will be a good improvement. Will change :+1: